plugins/pur/core/src/main/java/org/pentaho/di/repository/pur/PurRepository.java

```diff
@@ -237,12 +241,16 @@ protected enum RepositoryServers {
   private static final ReadWriteLock readWriteLock = new ReentrantReadWriteLock();
   private static final ReadWriteLock sharedObjectsLock = new ReentrantReadWriteLock();
 
+  private final MetastoreLocatorOsgi metastoreLocator;
+
 
   // ~ Constructors ====================================================================================================
 
-  public PurRepository() {
+  public PurRepository() throws KettlePluginException {
     super();
     initSharedObjectAssemblerMap();
+    Collection<MetastoreLocatorOsgi> metastoreLocators = PluginServiceLoader.loadServices( MetastoreLocatorOsgi.class );
```

Comment:
This is perfectly fine for POC, but did you ran into some problem looking for MetastoreLocator instead

Reply:
Yes using metastore locator instead of the osgi one required a new dependency that caused problems

Reply:
Oh yes, I remember now. I was able to look up MetastoreLocator and added dependency for pentaho-metastore-locator-api in pom.xml and it worked fine